apis/kubedb/v1alpha2/rabbitmq_webhook.go

```diff
@@ -0,0 +1,229 @@
+	rabbitmqlog.Info("validate update", "name", r.Name)
+	return nil, r.ValidateCreateOrUpdate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *RabbitMQ) ValidateDelete() (admission.Warnings, error) {
+	rabbitmqlog.Info("validate delete", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	var allErr field.ErrorList
+	if r.Spec.TerminationPolicy == TerminationPolicyDoNotTerminate {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("teminationPolicy"),
+			r.Name,
+			"Can not delete as terminationPolicy is set to \"DoNotTerminate\""))
+		return nil, apierrors.NewInvalid(schema.GroupKind{Group: "kafka.kubedb.com", Kind: "Kafka"}, r.Name, allErr)
+	}
+	return nil, nil
+}
+
+func (r *RabbitMQ) ValidateCreateOrUpdate() error {
+	var allErr field.ErrorList
+	// TODO(user): fill in your validation logic upon object creation.
+	if r.Spec.EnableSSL {
+		if r.Spec.TLS == nil {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("enableSSL"),
+				r.Name,
+				".spec.tls can't be nil, if .spec.enableSSL is true"))
+		}
+	} else {
+		if r.Spec.TLS != nil {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("enableSSL"),
+				r.Name,
+				".spec.tls must be nil, if .spec.enableSSL is disabled"))
+		}
+	}
+
+	// number of replicas can not be 0 or less
+	if r.Spec.Replicas != nil && *r.Spec.Replicas <= 0 {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("replicas"),
+			r.Name,
+			"number of replicas can not be 0 or less"))
+	}
+
+	err := r.validateVersion(r)
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("version"),
+			r.Name,
+			err.Error()))
+	}
+
+	err = r.validateVolumes(r)
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("podTemplate").Child("spec").Child("volumes"),
+			r.Name,
+			err.Error()))
+	}
+
+	err = r.validateVolumesMountPaths(&r.Spec.PodTemplate)
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("podTemplate").Child("spec").Child("volumeMounts"),
+			r.Name,
+			err.Error()))
+	}
+
+	if r.Spec.StorageType == "" {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+			r.Name,
+			"StorageType can not be empty"))
+	} else {
+		if r.Spec.StorageType != StorageTypeDurable && r.Spec.StorageType != StorageTypeEphemeral {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+				r.Name,
+				"StorageType should be either durable or ephemeral"))
+		}
+	}
+
+	if len(allErr) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(schema.GroupKind{Group: "kafka.kubedb.com", Kind: "Kafka"}, r.Name, allErr)
+}
+
+var rabbitmqAvailableVersions = []string{
+	"3.12",
+}
+
+func (r *RabbitMQ) validateVersion(db *RabbitMQ) error {
+	version := db.Spec.Version
```

Comment:
Instead of mainting a static list, dynamically get the version to check its existance.